src/BayesFilters/include/BayesFilters/SUKFCorrection.h

```diff
@@ -0,0 +1,68 @@
+#ifndef SUKFCORRECTION_H
+#define SUKFCORRECTION_H
+
+#include <BayesFilters/AdditiveMeasurementModel.h>
+#include <BayesFilters/GaussianCorrection.h>
+#include <BayesFilters/GaussianMixture.h>
+#include <BayesFilters/sigma_point.h>
+
+#include <Eigen/Dense>
+
+namespace bfl {
+    class SUKFCorrection;
+}
+
+/**
+ * This class implements the algorithm:
+ * Barfoot, T., McManus, C. (2011),
+ * 'A Serial Approach to Handling High-Dimensional Measurements in the Sigma-Point Kalman Filter.',
+ * Science and Systems VII,
+ * MIT Press
+ */
+
+class bfl::SUKFCorrection : public bfl::GaussianCorrection
+{
+public:
+    /**
+     * The input argument meas_sub_size is the sub-size, J, of the 
+     * measurement vector y in R^M such that M = k * J for some positive integer k.
+     */
+    SUKFCorrection(std::unique_ptr<AdditiveMeasurementModel> measurement_model, const std::size_t state_size, const double alpha, const double beta, const double kappa, const std::size_t measurement_sub_size) noexcept;
+
+    SUKFCorrection(SUKFCorrection&& sukf_correction) noexcept;
+
+    virtual ~SUKFCorrection() noexcept { };
+
+    MeasurementModel& getMeasurementModel() override;
+
+    /**
+     * If set to true, when the algorithm calls measurement_model_.getNoiseCovarianceMatrix()
+     * it expects to receive a noise covariance matrix of size measurement_sub_size_ x measurement_sub_size_
+     * and uses it for all the sub-vectors beloning to the current measurement.
+     * If set to false, the algorithms expects to receive the complete noise covariance matrix
+     * consisting in (M / meas_sub_size) diagonal blocks of size measurement_sub_size_ x measurement_sub_size_
+     * where M is the size of the current measurement.
+     */
+    void useReducedNoiseCovarianceMatrix(const bool value);
```

Comment:
I reckon that there should be at least one constructor where you can pass this parameter.
Otherwise a user should remember to call this function because the default value is `false`.
I'd like to have classes that are fully constructed by the constructor.